### PR TITLE
Telemetry improvements to get a funnel view for all save calls made to Dataverse

### PR DESCRIPTION
**Pull request by BidishaMS**
Need this data for creating Funnel view. 
1) How many save calls triggered?
2) How many save calls failed
3) How many save calls succeeded

### CODE REVIEW

File: src/web/client/dal/fileSystemProvider.ts [hunk 1]
```diff
@@ -198,9 +198,6 @@ export class PortalsFS implements vscode.FileSystemProvider {
                     title: vscode.l10n.t("Saving your file ..."),
                 },
                 async () => {
-                    WebExtensionContext.telemetry.sendInfoTelemetry(
```
Comment (tyaginidhi) [inline src/web/client/dal/fileSystemProvider.ts:201]: Do not remove this telemetry point - i see you have added it downstream - but what if there is a failure before this call is triggered - we will loose insight on failed save scenarios.

File: src/web/client/dal/remoteSaveProvider.ts [hunk 2]
```diff
@@ -157,7 +158,11 @@ async function saveDataToDataverse(
                 saveCallParameters.requestUrl,
                 entityName,
                 httpMethod.PATCH,
-                fileExtensionType
+                fileExtensionType,
+                undefined,
+                0,
+                undefined,
+                telemetryEventNames.WEB_EXTENSION_SAVE_FILE_TRIGGERED
```
Comment (tyaginidhi) [inline src/web/client/dal/remoteSaveProvider.ts:165]: I would suggest adding a new telemetry event here - one specific to dataverse save call triggered

File: src/web/client/dal/remoteSaveProvider.ts [hunk 3]
```diff
@@ -170,7 +175,9 @@ async function saveDataToDataverse(
                     entityName,
                     httpMethod.PATCH,
                     new Date().getTime() - requestSentAtTime,
-                    JSON.stringify(response)
+                    JSON.stringify(response),
+                    fileExtensionType,
+                    telemetryEventNames.WEB_EXTENSION_SAVE_FILE_FAILED
```
Comment (tyaginidhi) [inline src/web/client/dal/remoteSaveProvider.ts:180]: same here - the name could be dataverse save specific. Save of a file is "Save to VFS + Save to Dataverse" = success scenario

File: src/web/client/telemetry/webExtensionTelemetry.ts [hunk 4]
```diff
@@ -108,8 +110,9 @@ export class WebExtensionTelemetry {
         httpMethod: string,
         duration: number,
         entityFileExtensionType?: string, // TODO: Pass these as function properties parameters
+        eventName?:string
     ) {
-        this.sendAPITelemetry(URL, entity, httpMethod, entityFileExtensionType, true, duration, undefined, telemetryEventNames.WEB_EXTENSION_API_REQUEST_SUCCESS);
+        this.sendAPITelemetry(URL, entity, httpMethod, entityFileExtensionType, true, duration, undefined, !isNullOrUndefined(eventName)?eventName:telemetryEventNames.WEB_EXTENSION_API_REQUEST_SUCCESS);
```
Comment (tyaginidhi) [inline src/web/client/telemetry/webExtensionTelemetry.ts:115]: nit: spacing in ternary operator

File: src/web/client/telemetry/webExtensionTelemetry.ts [hunk 5]
```diff
@@ -119,8 +122,9 @@ export class WebExtensionTelemetry {
         duration: number,
         errorMessage?: string,
         entityFileExtensionType?: string, // TODO: Pass these as function properties parameters
+        eventName?:string
     ) {
-        this.sendAPITelemetry(URL, entity, httpMethod, entityFileExtensionType, false, duration, errorMessage, telemetryEventNames.WEB_EXTENSION_API_REQUEST_FAILURE);
+        this.sendAPITelemetry(URL, entity, httpMethod, entityFileExtensionType, false, duration, errorMessage, !isNullOrUndefined(eventName)?eventName:telemetryEventNames.WEB_EXTENSION_API_REQUEST_FAILURE);
```
Comment (tyaginidhi) [inline src/web/client/telemetry/webExtensionTelemetry.ts:127]: nit as above

**Review by tyaginidhi — CHANGES_REQUESTED** [review]

File: src/web/client/dal/remoteSaveProvider.ts (hunk 2 above)
Comment (BidishaMS) [reply src/web/client/dal/remoteSaveProvider.ts]: done

File: src/web/client/dal/remoteSaveProvider.ts (hunk 3 above)
Comment (BidishaMS) [reply src/web/client/dal/remoteSaveProvider.ts]: done

**Review by tyaginidhi — APPROVED** [review]

**Pull request merged**